packages/@aws-cdk/cx-api/lib/cloud-artifact.ts

```diff
@@ -19,6 +18,11 @@ export enum ArtifactType {
    * The artifact is an AWS CloudFormation stack.
    */
   AWS_CLOUDFORMATION_STACK = 'aws:cloudformation:stack',
+
+  /**
+   * The artifact contains metadata generated out of the CDK application.
+   */
+  CDK_METADATA = 'cdk:metadata',
```

Comment:
The modeling I had in mind was different. I modeled a generic 'cdk:metadata' type in the CloudAssembly and 'tree' is one instance of this metadata type.

Do you not like this style of modeling?

Reply:
The rest of your comments are artifacts of this difference in modeling choice.

Reply:
Yes, I think that's the crux of it. I don't see any value in modeling a generic type in the manifest. An artifact type in a cloud assembly is expected to a have a specific format/schema/use (otherwise, what can I do with it?). A generic "metadata" artifact doesn't have any meaning in my mind, so we might as well not model it at all. I am ok if we decide not to model this at all at this point and only add it later. Your call.